src/io/sampling.jl

```diff
@@ -350,6 +350,53 @@ function fix_factor!(d::Domain; factors...)::Nothing
     end
 end
 
+"""
+    set_factor_bounds!(d::Domain, factor::Symbol, new_bnds::Tuple)
+    set_factor_bounds!(d::Domain; factors...)
+
+Fix bounds of a parameter for sampling to those provided.
+
+Note: Changes are permanent. To reset, either specify the original value(s)
+      or reload the Domain.
+
+# Examples
+```julia
+# Fix `wave_stress` to specified bounds
+set_factor_bounds!(dom, :wave_stress, (0.1,0.2))
+```
+"""
+function set_factor_bounds!(d::Domain, factor::Symbol, new_bnds::Tuple)::Nothing
+    params = DataFrame(d.model)
+
+    # Check new parameter bounds are within old parameter bounds
+    if (new_bnds[1] < params[params.fieldname .== factor, :default_bounds][1][1]) ||
+        (new_bnds[2] > params[params.fieldname .== factor, :default_bounds][1][2])
+        error("New bounds should be within default bounds.")
+    end
+    if (params[params.fieldname .== factor, :dists][1] == "triang") &&
+        (length(new_bnds) !== 3)
+        error("Triangular dist requires three parameters.")
+    end
+    if (params[params.fieldname .== factor, :dists][1] == "unif") &&
+        (length(new_bnds) !== 2)
+        error("Uniform dist requires two parameters.")
+    end
+
+    params[params.fieldname.==factor, :bounds] .= [new_bnds]
+    params[params.fieldname .== factor, :val] .=
+        new_bnds[1] + 0.5 * (new_bnds[2] - new_bnds[1])
+
```

Comment:
Won't this be an issue if the factor whose bounds are being set is categorical/integer?

Reply:
Just added a check for integer/categoricals